### PR TITLE
feat(react): migrate webpack config

### PR DESCRIPTION
**Pull request by mandarini**
Similar to https://github.com/nrwl/nx/pull/14607, but for React projects.

### CODE REVIEW

File: packages/react/src/migrations/update-15-6-3/webpack-config-setup.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          options['webpackConfig'],
+          `
+            const { composePlugins, withNx } = require('@nrwl/webpack');
+            const { withReact } = require('@nrwl/react');
+
+            // Nx plugins for webpack.
+            module.exports = composePlugins(withNx(), withReact(), (config, { options, context }) => {
+              // Note: This was added by an Nx migration.
+              // You should consider inlining the logic into this file.
+              return require('./${justTheFileName}')(config, context);
```
Comment (jaysoo) [inline packages/react/src/migrations/update-15-6-3/webpack-config-setup.ts:58]: should this be `oldName`? The original file is overwritten.

File: packages/react/src/migrations/update-15-6-3/webpack-config-setup.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        projectConfiguration.targets[targetName].options = options;
+        updateProjectConfiguration(tree, projectName, projectConfiguration);
+      }
+    }
+  );
+
+  await formatFiles(tree);
+}
+
+function renameFile(tree: Tree, from: string, to: string) {
```
Comment (jaysoo) [inline packages/react/src/migrations/update-15-6-3/webpack-config-setup.ts:121]: For later: It'd be nice to add something like this to devkit.

File: packages/react/src/migrations/update-15-6-3/webpack-config-setup.ts (hunk 1 above)
Comment (mandarini) [reply packages/react/src/migrations/update-15-6-3/webpack-config-setup.ts:58]: Yes!!!! I saw that too, oops!!! 😅

Comment (mandarini) [reply packages/react/src/migrations/update-15-6-3/webpack-config-setup.ts:58]: The issue is actually above, on the line where I'm setting the `justTheFileName`. It's using the `webpackConfig` path, not the `oldName` to get the file name. I fixed that line,

Comment (jaysoo) [reply packages/react/src/migrations/update-15-6-3/webpack-config-setup.ts:58]: oh i get it, sorry i totally misread

**Review by jaysoo — APPROVED** [review]

**Pull request merged**